deep_reference_parser/split.py

```diff
@@ -156,11 +167,30 @@ def split(self, text, return_tokens=False, verbose=False):
     text=("Plaintext from which to extract references", "positional", None, str),
     config_file=("Path to config file", "option", "c", str),
     tokens=("Output tokens instead of complete references", "flag", "t", str),
-    verbose=("Output more verbose results", "flag", "v", str),
+    outfile=("Path to json file to which results will be written", "option", "o", str),
 )
-def predict(text, config_file=LATEST_CFG, tokens=False, verbose=False):
-    predictor = Predictor(config_file)
-    out = predictor.split(text, return_tokens=tokens, verbose=verbose)
+def split(text, config_file=SPLITTER_CFG, tokens=False, outfile=None):
+    """
+    Runs the default splitting model and pretty prints results to console unless
+    --outfile is parsed with a path. Can output either tokens (with -t|--tokens)
+    or split naively into references based on the b-r tag (default).
+
+    NOTE: that this function is provided for examples only and should not be used
+    in production as the model is instantiated each time the command is run. To
+    use in a production setting, a more sensible approach would be to replicate
+    the split or parse functions within your own logic.
+    """
+    splitter = Splitter(config_file)
+    if outfile:
+        out = splitter.split(text, return_tokens=tokens, verbose=False)
+
+        try:
+            with open(outfile, "w") as fb:
+                json.dump(out, fb)
+            msg.good(f"Wrote model output to {outfile}")
+        except:
+            msg.fail(f"Failed to write output to {outfile}")
+
+    else:
+        out = splitter.split(text, return_tokens=tokens, verbose=True)
```

Comment:
I think this line runs in both cases of the if then else so could run before

Reply:
the difference is whether it is verbose or not. One will print to console, the other will not